### ISSUE TITLE
[MJAVADOC-420] javadoc:fix duplicates lines

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#718]
**[Michenaud Laurent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michenux)** opened **[MJAVADOC-420](https://issues.apache.org/jira/browse/MJAVADOC-420?redirect=false)** and commented

On the following example, if you starts several times the javadoc:fix goal, the line "`@param` paramSecond" is duplicated several times.

```
package com.example;

/**
 * <p>FirstBug class.</p>
 */
public class FirstBug {

	/**
	 * <p>mymethod.</p>
	 *
	 * @param param a {@link java.lang.String} object.
	 * @param paramSecond a {@link java.lang.String} object.
	 */
	public void mymethod( String param, String paramSecond ) {
		
	}
}
```

After investigation, i have found that when a parameter name is included in another parameter name, the bug occurs.

Best regards
Laurent

---

**Affects:** 2.10.1

**Attachments:**
- [fixjavadocbugs.zip](https://issues.apache.org/jira/secure/attachment/12717023/fixjavadocbugs.zip) (_4.17 kB_)

1 votes, 3 watchers

**Comment by jira-importer** [comment]
**[Richard Sand](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rsand)** commented

FYI I provided a patch for this issue attached to MJAVADOC-452

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [28db4b08fc030036a138c36363b0fd2540a99e5f](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git;a=commit;h=28db4b08fc030036a138c36363b0fd2540a99e5f)
